src/annotator/plugin/document.coffee

```diff
@@ -168,3 +170,14 @@ module.exports = class Document extends Plugin
     d = document.createElement('a')
     d.href = url
     d.href
+
+# get the true URI record when it's masked via a different protocol.
+# this happens when an href is set with a uri using the 'blob:' protocol
+# but the document can set a different uri through a <base> tag.
+
+  _getDocumentHref: ->
+    href = document.location.href
+    if new URL(href).protocol != new URL(baseURI).protocol
```

Comment:
I wonder if there are any sites out there that include a `<base>` tag with an HTTP URL when the document is served over HTTPS or vice-versa, in which case this could return an unexpected URL.